### PR TITLE
New method `AddParameter` with index in `MutableExecutableDeclaration`

### PR DESCRIPTION
**Pull request by dvojtise**
This contributes a new API method in `MutableExecutableDeclaration` (and `JvmExecutableDeclarationImpl`) for active annotation processing.

It allows adding a parameter to a method declaration by specifying the position via an index.

The PR includes a test suite illustrating a minimal use case. It checks inserting at the beginning, in the middle of the parameter list and also checks that it raises an IndexOutOfBoundsException if the index is invalid.

This fixes https://github.com/eclipse/xtext/issues/2777

### CODE REVIEW

File: org.eclipse.xtend.core/xtend-gen/org/eclipse/xtend/core/macro/declaration/JvmExecutableDeclarationImpl.java [hunk 1]
```diff
@@ -140,6 +140,22 @@ public MutableParameterDeclaration addParameter(final String name, final TypeRef
     return ((MutableParameterDeclaration) _parameterDeclaration);
   }
 
+  public MutableParameterDeclaration addParameter(final int index, final String name, final TypeReference type) {
```
Comment (szarnekow) [inline org.eclipse.xtend.core/xtend-gen/org/eclipse/xtend/core/macro/declaration/JvmExecutableDeclarationImpl.java:143]: Would it make sense to delegate from `addParameter(String, TypeReference)` to this new method?

**Comment by LorenzoBettini** [comment]
@dvojtise @szarnekow what's the status of this?

**Comment by dvojtise** [comment]
I've taken into account all comments, for me it could me merged

**Review by szarnekow — APPROVED** [review]
Thank you for the pull request. Appreciated.

**Comment by LorenzoBettini** [comment]
@dvojtise can it be you forgot to commit generated files? I see some changes on my workspace in `/org.eclipse.xtend.core/xtend-gen/org/eclipse/xtend/core/macro/declaration/JvmExecutableDeclarationImpl.java`

**Comment by dvojtise** [comment]
> @dvojtise can it be you forgot to commit generated files? I see some changes on my workspace in `/org.eclipse.xtend.core/xtend-gen/org/eclipse/xtend/core/macro/declaration/JvmExecutableDeclarationImpl.java`

mmm, maybe yes, my IDE automatically added a .gitignore in all xtend-gen sub folders so I didn't checked them  in detail.

**Comment by LorenzoBettini** [comment]
@dvojtise that Git option is so dangerous that it's the first thing I disable ;)

I thought our Oomph disables that... had you Oomphed the Xtext workspace?

**Comment by pbrossel** [comment]
> @dvojtise that Git option is so dangerous that it's the first thing I disable ;)
> 
> I thought our Oomph disables that... had you Oomphed the Xtext workspace?

Can you tell me where to disable this ? I would need the info for another project where I want the generated sources to be committed.

**Comment by LorenzoBettini** [comment]
@dvojtise IIRC, preferences, git, project; there you find an option for ignoring generated resources

**Pull request merged**